### PR TITLE
More correct media type handling

### PR DESCRIPTION
**Pull request by TimothyGu**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -29,24 +29,35 @@ function dataUriToBuffer (uri) {
...
-    } else if (0 == meta[i].indexOf('charset=')) {
-      charset = meta[i].substring(8);
+    } else {
+      typeFull += ';' + meta[i];
+      if (0 == meta[i].indexOf('charset=')) {
+        charset = meta[i].substring(8);
+      }
     }
   }
+  // defaults to US-ASCII only if type is not provided
```
Comment (TooTallNate) [inline index.js:46]: What's the thought on this API change? We're still defaulting the type to `text/plain` so why not have a default charset as well?

**Comment by TooTallNate** [comment]
This seems very nice @TimothyGu!

**Comment by TimothyGu** [comment]
In Chrome at least, `data:text/plain,%E4%B8%AD%E6%96%87` ("中文" UTF-8 encoded) outputs "中文", yet `data:,%E4%B8%AD%E6%96%87` does not. This suggests that if main type is specified, the charset default is not used.

Indeed, the spec ([RFC 2397](https://tools.ietf.org/html/rfc2397)) only defines special cases with no type and with charset only (defaults to `text/plain`). If only the main type is specified, the main type should be treated as the entire type string.

BTW we are trying to add `data:` support to [node-fetch](https://github.com/bitinn/node-fetch) and this module seems to be a good fit!

**Comment by TooTallNate** [comment]
Effectively it just makes the default be `utf8` since that's what Node.js Buffer constructor default encoding is, but since that seems to match Chrome's behavior then that sounds like a good justification. That means we can treat it more like a bug fix than a breaking change.

**Comment by TooTallNate** [comment]
Beautiful pull request by the way. Thank you!

**Comment by TimothyGu** [comment]
@TooTallNate, no, we are not actually defaulting on anything. We percent decode into a Buffer without caring about what the bytes mean.

**Comment by TimothyGu** [comment]
And thanks for merging :)

**Comment by TooTallNate** [comment]
> We percent decode into a Buffer without caring about what the bytes mean.

Totally right. I guess it's just Chrome's rendering of it that is UTF-8 by default.

**Comment by TooTallNate** [comment]
Published in v1.1.0.

**Pull request merged**